R/check_input.R

```diff
@@ -335,7 +335,10 @@ check_oneof_input <- function(input, property = c("required", "optional"), # nol
 
   if (typeof(input) == "character") {
     value_schema <- oneof_schema[["character"]]
-    if (!any((grepl(value_schema[["pattern"]], input)))) {
+    if (!is.null(value_schema[["pattern"]]) && !any(
+      grepl(value_schema[["pattern"]], input)
+    )
+    ) {
```

Comment:
This if statement threw me off a bit because I was wondering if there was an extra brace. I've taken the Jenny Bryan route and put it into an indicator variable. 

```suggestion
    unmatched_values_exist <- !is.null(value_schema[["pattern"]] &&
      !any(grepl(value_schema[["pattern"]], input))
    if (unmatched_values_exist) {
```

Reply:
Jenny Bryan for the win! Missed a sneaky bracket though 😉. Fixed and will merge as soon as tests pass 😁

Reply:
NOOOOOOOOO! So embarass. Much shame. 🙈